src/include/sof/audio/component.h

```diff
@@ -18,30 +18,13 @@
 
 #include <sof/audio/buffer.h>
 #include <sof/audio/format.h>
-#include <sof/audio/pipeline.h>
-#include <rtos/panic.h>
 #include <rtos/idc.h>
-#include <sof/common.h>
-#include <sof/list.h>
-#include <rtos/alloc.h>
-#include <sof/lib/cpu.h>
 #include <sof/lib/dai.h>
-#include <sof/lib/memory.h>
-#include <sof/lib/perf_cnt.h>
-#include <sof/math/numbers.h>
 #include <sof/schedule/schedule.h>
-#include <rtos/sof.h>
-#include <sof/trace/trace.h>
 #include <ipc/control.h>
-#include <ipc/stream.h>
 #include <sof/ipc/topology.h>
 #include <kernel/abi.h>
-#include <user/trace.h>
 
-#include <errno.h>
-#include <stdbool.h>
-#include <stddef.h>
-#include <stdint.h>
```

Comment:
I'm very suspicious about all these removals, how did you figure out, that all these headers aren't needed?

Reply:
I removed headers one by one and check build on sof main. I don't know why there a so many headers not being used or included somewhere else and being redundantly called.

Reply:
This isn't the criterion how we decide whether a header is needed in a file or not. For every feature used in a file, if that feature requires a header, we include that header explicitly in that file instead of relying on indirect inclusion. I can think of two exceptions to this rule:
1. For structure pointers a forward declaration of that structure is enough
2. Some headers explicitly must only be included in a single header that wraps them. This is typical for platform files e.g. src/platform/meteorlake/include/platform/lib/cpu.h

Reply:
> I removed headers one by one and check build on sof main.

This proves nothing because they are frequently included indirectly still. For real solutions check

- https://github.com/zephyrproject-rtos/zephyr/issues/41543

Reply:
The full CI build of FW, tooling,docs etc should pick up any issues here, so I would say all green CI would mean header cleanups are good fo current codebase. However, we could get lucky in some places and have to manually update some headers with dependencies as we find errors.

Reply:
@lgirdwood sorry, I humbly disagree. We could have a single header, that include all other imaginable headers, and have all .c files just include that single header. That would probably pass all build- and run-time tests (with some tweaking probably). I don't think that's what we want.

Reply:
We had this discussion before with @mmaka1 and @marcinszkudlinski  and creating monster header with all headers is not good idea.

Reply:
@lyakh I agree, I'm not sure we can remove all the POSIX headers, i.e. I'm pretty sure we use stdint.h for all structure types, but the other SOF API headers, if they are NOT used in that header file then we can remove (and yes it wont be 100% test build-able as we cant yet independently build each header on its own).  Forward structure/type declarations will no doubt be needed too.

What I mean here is we have a CI pass meaning we are build testing the best we can today and this is not a proof that all headers are build-able after this change (if that is even true before this PR today), but an indication that we have some confidence. 

I think we all agree that we are not 
1) Doing monster header files.
2) Duplicating header files in multiple locations.
3) Creating derived header files with copy and paste structure and API definitions (but we can move structures to align with "needed by modules" rule).